services/common/src/main/java/org/finra/gatekeeper/common/services/user/auth/GatekeeperAuthorizationService.java

```diff
@@ -75,7 +77,11 @@ public Optional<Set<String>> load(String userName) throws Exception {
     }
 
     public Set<String> getMemberships(){
-        return userMembershipCache.getUnchecked(gatekeeperUserProfileSupplier.get().getName()).get();
+        return useHeadersForMembership ? getMembershipsFromHeader() : userMembershipCache.getUnchecked(gatekeeperUserProfileSupplier.get().getName()).get();
+    }
+
+    private Set<String> getMembershipsFromHeader(){
+        return gatekeeperUserProfileSupplier.get().getMemberships();
```

Comment:
This should not be implemented as part of this class, this should be implemented within a class that extends GatekeeperAuthorizationService